dask/utils.py

```diff
@@ -843,6 +843,8 @@ def infer_storage_options(urlpath, inherit_storage_options=None):
     if protocol == 'file':
         # Special case parsing file protocol URL on Windows according to:
         # https://msdn.microsoft.com/en-us/library/jj710207.aspx
+        # urlsplit does not work with ?
+        path = urlpath if "?" in urlpath else path
```

Comment:
This sounds like a dealbreaker, if true!

Also, I've been thinking, that the inclusion of * and other characters doesn't necessarily mean that the string is for globbing, because the character may have been escaped. The translation of glob-string to regex in s3fs and hdfs3 will probably break in those cases. 
For glob.glob, of course we expect the normal rules to be followed exactly. `glob.has_magic`, used by glob.glob, checks for regex `'([*?[])'`; the translation of glob patterns to regex is done by fnmatch.translate (and filtered in fnmatch.filter).

We could choose to limit the use of glob-strings to the simple case here, and force users to filter their own file-lists for anything more complex.